intake/readers/readers.py

```diff
@@ -360,7 +360,7 @@ def _duck(self, data):
             # https://duckdb.org/docs/extensions/
             if conn.startswith("sqlite:"):
                 duckdb.connect(":default:").execute("INSTALL sqlite;LOAD sqlite;")
-                conn = re.sub("^sqlite3?://", "", conn)
+                conn = re.sub("^sqlite3?(:///|://|:/)", "", conn)
```

Comment:
```suggestion
                conn = re.sub("^sqlite3?:[/]+", "", conn)
```
perhaps?

Reply:
Accordin to the [URI syntax](https://datatracker.ietf.org/doc/html/rfc3986#section-3), only the following combinations are possible:

1. `sqlite://file:/path/to/database.db` with an authority component `file:`.
2. `sqlite:///path/to/database.db` with an empty authority component.
3. `sqlite:/absolute/path/to/database.db` without an authority component, the `/` after the `:` symbol points to the root of the file system.
4. `sqlite:relative/path/to/database.db` without an authority component, the absence of the `/` after the `:` symbol means that the path is relative to the current folder.

From [the documentation](https://datatracker.ietf.org/doc/html/rfc3986#section-3.3):

> If a URI contains an authority component, then the path component must either be empty or begin with a slash ("/") character.

Therefore, a path with more than three slashes after the `:` symbol is not allowed.
Please correct me if I'm wrong.

I would prefer a code that breaks [as soon as possible](https://en.wikipedia.org/wiki/Fail-fast_system), instead of a code that silently corrects an invalid input and makes the debugging harder.

Reply:
Nice suggestion! It seems more readable to me.